### ISSUE TITLE
Modern UI/UX for RGOV interfaces

### ISSUE THREAD

**Issue by David405** [rchain-community/rgov#204]
One thing for me that has sort of acted as a blocker in implementing a new UI for the voting interface has been the design standards that should be followed, material UI? Bootstrap?? What color codes, font types, UI elements.

Also, another reason for creating this issue was as a result of the discussions from issue #201 and pull request #190. 

So, I would really love input from everyone here so the UI team and development can move alot faster without having conflicting designs

**Comment by w2vy** [comment]
I think there are two goals that aee almost at odds with each other.
On the one hand I think we need to have a clean interface that is something a non-programmer would use but at the same time we like to show the Rholang contracts that are being signed as a "Proof of Design' (Yes we are really performing on chain contracts)

So I wonder if we could have an interface where I login, do various things (show issues, castVote etc) and have on each page a Developer Button to toggle a Show Rholang window

In order to design each web page that would be displayed for each interaction we need to define a list of roles and interactions.

For example there is always a Vote Organizer and a set of Voters; so before we can define the script we use for the next demo we need to define the operations each role can perform as well as dependencies and additional inputs

Organizer can: Create an Issue/Ballot, Add Voters/Groups - What else?
Voters can: View Issue/Ballot, CastVote, DelegateVote, TallyVote - What else?

I think @jimscarver may be the best to expand that list as I believe he laid the foundation of the design.
Let's keep it to atomic functions it may be as simple as going through all the actions and pick a subset that should be used in the next voting demo and what capabilities we want to demonstrate

**Comment by dckc** [comment]
Please narrow the score of this issue.

If it's about visual design, propose some visual elements.

If it's about dependencies, make a proposal to add one or more dependencies and explain how  the dependency helps address project requirements.

#201 is about being explicit about taking on dependencies. Each new dependency is judged on it's own costs and benefits. I can evaluate a proposal to use material or a proposal to use bootstrap, if the proposal is clear about the benefits. This is easiest if the proposal is s PR that addresses some issue that we have agreed needs assessing.

The problems with #190 were
  1. Commits with diffs that didn't match their commit message
  2. Commits that went beyond the scope of the PR
 3. Not following existing standards from CONTRIBUTING.md

**Comment by dckc** [comment]
Oh..  Another aspect of #190 is that syntax highlighting enhancement #134 isn't clearly related to the goal of scalable governance. It's not on any agreed milestone. So review of a PR for it should be quite conservative. I would rather postpone the enhancement than sacrifice a clean git history, ocap discipline, etc.

**Comment by jimscarver** [comment]
We have embraced the developer interface since community involvement is necessary for the evolution of decentralized governance. https://rushkoff.com/books/program-or-be-programmed/

The syntax highlighting will significantly enhance the developer interface nut we do have other priorities. At the same time we want to encourage people to work where their passion is.

I do not like the split editing screen. Finding the result becomes unobvious. I love davids new styles and fixing editing would be worthwhile in my view.

Design issues may not be relevant if this is a "standards" thread but we do need to consent to our design directions. Nobody gets their way in a cooperation.

**Comment by dckc** [comment]
> ... I love davids new styles ...

pointer?

**Comment by jimscarver** [comment]
> > ... I love davids new styles ...
> 
> pointer?

https://deploy-preview-190--rgov1.netlify.app/

**Comment by David405** [comment]
@w2vy when you say:

> show the Rholang contracts that are being signed as a "Proof of Design' (Yes we are really performing on-chain contracts)

do you mean the developer interface or the voting interface?

**Comment by David405** [comment]
@dckc @jimscarver the whole point of the issue is to get everyone's thought as to what set of tools ( dependencies, libraries) to use, if we have to create an issue for each separate library or dependency, don't you think that would rather slow down the process?

**Comment by dckc** [comment]
I don't see any way to anticipate all the library dependencies we will ever need. Each new dependency must be motivated by some feature, and a new issue for each feature is completely ordinary, no?

**Comment by David405** [comment]
@dckc so this means for any new library or dependency that we may add, we have to create a new issue? I am trying to gain clarity.

**Comment by dckc** [comment]
All development is in the context of some issue. Every line of code should be in service to some issue. So yes, since adding a dependency involves lines of code, it should be in service to some issue.

If interactjs were needed for syntax highlighting, then we wouldn't need #201 separate from #134 . But the stated motivation for interactjs is not to make syntax highlighting work, but to change the way textarea resizing works. That's a separate issue, to my mind.

**Comment by dckc** [comment]
Issues / PRs might have a wider scope, such as "modern web design for actions UI". That's sometimes more straightforward, sometimes less than more tightly scoped issues. It's more straightforward if everybody's on the same page based on previous discussion.

**Comment by Bill-Kunj** [comment]
@David405 is asking (I think) about what we want as a (somewhat) standard web experience.
I think bootstrap is a convenient, easy way to proceed to a more inviting RGOV experience, and I think we should solicit @David405 advice since @David405 has the knowledge and expertise RGOV lacks.
Let's let the work happen and see what results.
#My2c

**Comment by Bill-Kunj** [comment]
The issues raised here are:
Not blocking to *RGOV* development
Not blocking to *RGOV* deployment

Let's bring the details in other issues?

Closing
